### ISSUE TITLE
Should `role-supports-aria-props` fail when role is `checkbox` and `aria-required` is set?

### ISSUE THREAD

**Issue by acostalima** [jsx-eslint/eslint-plugin-jsx-a11y#716]
ARIA's specification for [`aria-required`](https://www.w3.org/TR/wai-aria/#aria-required) leads me to believe so but I'm not an expert in the matter. The following example was failing on v6.2.3 and on v6.3.1 it's not anymore:

```html
<span role="checkbox" aria-checked="false" aria-required />
```

**Comment by jessebeach** [comment]
@acostalima it was failing because we didn't have the prop explicitly listed in our allowed props. As of the new version it is.

Unlike HTML, the _presence_ of an ARIA attribute does not endow it with a truthy value. You need to explicitly assign it a value of `"true"` or `"false"`:

>__True/False__: Value representing either true or false. The default value for this value type is false unless otherwise specified.
--https://www.w3.org/TR/wai-aria/#valuetype_true-false

Keep in mind that an individual checkbox can be required, but not an individual radio. Only a radiogroup can be required: https://www.w3.org/TR/wai-aria-1.2/#radiogroup

**Comment by acostalima** [comment]
@jessebeach thanks for the clarification!
